app/src/main/java/de/westnordost/streetcomplete/quests/postbox_collection_times/AddCollectionTimesForm.kt

```diff
@@ -69,8 +80,21 @@ class AddCollectionTimesForm : AbstractQuestFormAnswerFragment<CollectionTimesAn
         }
     }
 
-    private fun loadCollectionTimesData(savedInstanceState: Bundle?): List<WeekdaysTimesRow> =
-        savedInstanceState?.let { Json.decodeFromString(it.getString(TIMES_DATA)!!) } ?: listOf()
+    private fun initStateFromTags() {
+        val ct = osmElement!!.tags["collection_times"]
+        val rows = ct?.toOpeningHoursRules()?.toOpeningHoursRows()
```

Comment:
There should be a function called `OpeningHoursRuleList.toCollectionTimesRows()` that both checks if it is really valid collection times and also if it is supported by streetcomplete. F.e. months, off days etc etc are not supported for collection times.

Also that function should return a data structure that is fitting for collection times, f.e. `WeekdaysTimesRow` As said, those are not time ranges.